sqldelight-idea-plugin/src/main/kotlin/app/cash/sqldelight/intellij/run/RunSqlAction.kt

```diff
@@ -0,0 +1,131 @@
+package app.cash.sqldelight.intellij.run
+
+import app.cash.sqldelight.core.compiler.model.NamedMutator
+import app.cash.sqldelight.core.compiler.model.NamedQuery
+import app.cash.sqldelight.core.lang.SqlDelightFile
+import app.cash.sqldelight.core.lang.SqlDelightFileType
+import app.cash.sqldelight.core.lang.psi.StmtIdentifierMixin
+import app.cash.sqldelight.core.lang.util.findChildOfType
+import app.cash.sqldelight.core.lang.util.range
+import app.cash.sqldelight.core.lang.util.rawSqlText
+import app.cash.sqldelight.core.psi.SqlDelightStmtClojure
+import app.cash.sqldelight.core.psi.SqlDelightStmtClojureStmtList
+import com.alecstrong.sql.psi.core.psi.SqlStmt
+import com.alecstrong.sql.psi.core.psi.SqlStmtList
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.application.runReadAction
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiComment
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFileFactory
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.psi.util.PsiTreeUtil
+import org.jetbrains.annotations.VisibleForTesting
+
+@VisibleForTesting
+internal class RunSqlAction(
+  private val stmt: SqlStmt,
+  private val project: Project = stmt.project,
+  private val executor: SqlDelightStatementExecutor = SqlDelightStatementExecutor.getInstance(
+    project
+  ),
+  private val dialogFactory: ArgumentsInputDialog.Factory = ArgumentsInputDialogFactoryImpl()
+) : AnAction() {
+
+  override fun actionPerformed(e: AnActionEvent) {
```

Comment:
I assume this would be visible along with the android studio action that we just merged in - the implementations are actually so similar it seems like this `AnAction` stuff might be able to just detect if android is around and use that to connect vs the custom jdbc stuff